lib/ansible/modules/network/nxos/nxos_linkagg.py

```diff
@@ -135,7 +135,7 @@
 
 def search_obj_in_list(group, lst):
     for o in lst:
-        if o['group'] == group:
+        if str(o['group']) == str(group):
```

Comment:
Looks like this is because of https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_linkagg.py#L267. We did it because some of the platforms/versions return group as `str`. 

Instead of masking it here, I would rather change the `type` of `group` param to `str` https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_linkagg.py#L375 and make sure the that the group info we fetch from the device is string format `obj['group'] = str(group)` https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_linkagg.py#L337

Reply:
Agreed, I'll change and retest.